src/modules/previewpane/powerpreview/powerpreview.cpp

```diff
@@ -75,28 +75,43 @@ void PowerPreviewModule::set_config(const wchar_t* config)
     }
 }
 
-// Enable the powertoy
+// Enable preview handlers.
 void PowerPreviewModule::enable()
 {
-    m_prevPaneSVGSettings.EnablePreview();
-    m_prevPaneMDSettings.EnablePreview();
-    Trace::FilePreviewerIsEnabled();
+    for (FileExplorerPreviewSettings& previewHanlder : this->m_previewHandlers)
+    {
+        previewHanlder.EnablePreview();
+    }
     this->m_enabled = true;
 }
 
-// Disable the powertoy
+// Disable all preview handlers.
 void PowerPreviewModule::disable()
 {
-    m_prevPaneSVGSettings.DisablePreview();
-    m_prevPaneMDSettings.DisablePreview();
-    Trace::FilePreviewerIsDisabled();
+    for (FileExplorerPreviewSettings& previewHanlder : this->m_previewHandlers)
+    {
+        previewHanlder.DisablePreview();
+    }
     this->m_enabled = false;
 }
 
 // Returns if the powertoys is enabled
 bool PowerPreviewModule::is_enabled()
 {
-    return this->m_enabled;
+    for (FileExplorerPreviewSettings& previewHanlder : this->m_previewHandlers)
```

Comment:
I think the Enable() code should be modified to handle this while this function remains as this->m_enabled. Based on the name, the is_enabled() function shouldn't be setting the m_enabled state.

Reply:
I am not enabling in the is_enabled() method. Is that what you meant? See the comment on the method:

        // if : at least one preview handler is enabled.
        //      => set the General settings state for the preview handlers to true.
        // if : No preview handler is enabled.
        //      => set the General settings state for preview hanlders to false.

Reply:
Based on the latest changes, I think is_enabled() can be reverted to `return this->m_enabled;` right?

Reply:
yes sure.